rest/route/distro.go

```diff
@@ -386,9 +415,12 @@ func (h *distroAMIHandler) Parse(ctx context.Context, r *http.Request) error {
 }
 
 func (h *distroAMIHandler) Run(ctx context.Context) gimlet.Responder {
-	d, err := data.FindDistroById(h.distroID)
+	d, err := distro.FindOne(distro.ById(h.distroID))
```

Comment:
Should check if distro is nil

Reply:
distro.FindOne was a bit different from the other model FindOne functions in that it has no check for `if adb.ResultsNotFound(err)` in it, so a nil distro would also cause an error to return.  There are quite a few places where this function is used not touched by this PR, I was concerned that making a nil distro no longer an error in the DB function might cause unexpected panics/errors elsewhere where we rely on our distro DB result not being nil.

That being said I can change the distro DB functions and their usages to be more in line with how tasks/hosts/etc handle it but that may be out of scope for this PR since it addresses a different tech debt. Happy to create a ticket for it!

Reply:
Sorry I actually addressed this in a different comment I can't find haha but yeah a different ticket seems reasonable!